verification/curator-service/ui/src/components/LinelistTable.tsx

```diff
@@ -219,6 +222,81 @@ function SearchBar(props: {
     );
 }
 
+const rowMenuStyles = makeStyles((theme: Theme) => ({
+    menuItemTitle: {
+        marginLeft: theme.spacing(1),
+    },
+}));
+
+function RowMenu(props: {
+    rowId: string;
+    setError: (error: string) => void;
+    refreshData: () => void;
+}): JSX.Element {
+    const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+    const history = useHistory();
+    const classes = rowMenuStyles();
+
+    const handleClick = (event: React.MouseEvent<HTMLButtonElement>): void => {
+        event.stopPropagation();
+        setAnchorEl(event.currentTarget);
+    };
+
+    const handleClose = (event?: any): void => {
+        event?.stopPropagation();
+        setAnchorEl(null);
+    };
+
+    const handleEdit = (event?: any): void => {
+        event?.stopPropagation();
+        history.push(`/cases/edit/${props.rowId}`);
+    };
+
+    const handleDelete = async (event?: any): Promise<void> => {
+        event?.stopPropagation();
+        try {
+            props.setError('');
+            const deleteUrl = '/api/cases/' + props.rowId;
+            await axios.delete(deleteUrl);
+            handleClose();
+            props.refreshData();
+        } catch (e) {
+            props.setError(e.toString());
+            handleClose();
+        }
+    };
+
+    return (
+        <>
+            <IconButton
+                aria-controls="topbar-menu"
+                aria-haspopup="true"
+                aria-label="row menu"
+                data-testid="row menu"
+                onClick={handleClick}
+                color="inherit"
+            >
+                <MoreVertIcon />
+            </IconButton>
+            <Menu
+                anchorEl={anchorEl}
+                keepMounted
+                open={Boolean(anchorEl)}
+                onClose={handleClose}
+            >
+                <MenuItem onClick={handleEdit}>
```

Comment:
Instead of having to deal with history and pushing stuff to it, we should simply use <Link> elements, I think I've done that in the main menu or somewhere else... I've read it's better practice :) (it is definitely simpler)

Reply:
In this case I don't think it's worth it to use `<Link>`. It would only be useful on the Edit MenuItem since the delete MenuItem isn't doing a page change. It also can't be used for navigating to Details since we just get an onRowClicked callback to trigger that. For the edit menu item we still need to handle onClick on the MenuItem because we need to stop the event propagation from triggering the row click. Since we already need that it's just two more lines to `useHistory` and call `history.push(...)`